discord/poll.py

```diff
@@ -0,0 +1,624 @@
+            else:
+                poll_media['emoji'] = {'name': str(emoji)}  # type: ignore
+
+        payload: PollAnswerWithIDPayload = {'answer_id': id, 'poll_media': poll_media}
+
+        return cls(data=payload, message=message, poll=poll)
+
+    @property
+    def text(self) -> str:
+        """:class:`str`: Returns this answer display text."""
+        return self.media.text
+
+    @property
+    def emoji(self) -> Optional[Union[PartialEmoji, Emoji]]:
+        """Optional[:class:`PartialEmoji`]: Returns this answer display emoji, is any."""
+        if isinstance(self.media.emoji, str):
+            return PartialEmoji.from_str(self.media.emoji)
+        return self.media.emoji
+
+    @property
+    def vote_count(self) -> int:
+        """:class:`int`: Returns an approximate votes of this answer. If the poll
+        has finalized, it returns the exact amount of votes.
+        """
+        return self._vote_count
+
+    @property
+    def poll(self) -> Poll:
+        """:class:`Poll`: Returns the parent poll of this answer"""
+        return self._poll
+
+    def _to_dict(self) -> PollAnswerPayload:
+        return {
+            'poll_media': self.media.to_dict(),
+        }
+
+    async def voters(
+        self, *, limit: Optional[int] = None, after: Optional[Snowflake] = None
+    ) -> AsyncIterator[Union[User, Member]]:
+        """Returns an :term:`asynchronous iterator` representing the users that have voted to this answer.
+
+        The ``after`` parameter must represent a user
+        and meet the :class:`abc.Snowflake` abc.
+
+        .. note::
+
+            This can only be called when the parent poll has an attached message.
+
+        Examples
+        --------
+
+        Usage ::
+
+            async for voter in poll_answer.voters():
+                print(f'{voter} has voted for {poll_answer}!')
+
+        Flattening into a list: ::
+
+            voters = [voter async for voter in poll_answer.voters()]
+            # voters is now a list of User
+
+        Parameters
+        ----------
+        limit: Optional[:class:`int`]
+            The maximum number of results to return.
+            If not provided, returns all the users who
+            voted to this poll answer.
+        after: Optional[:class:`abc.Snowflake`]
+            For pagination, voters are sorted by member.
+
+        Raises
+        ------
+        HTTPException
+            Retrieving the users failed.
+
+        Yields
+        ------
+        Union[:class:`User`, :class:`Member`]
+            The member (if retrievable) or the user that has voted
+            to this poll answer. The case where it can be a :class:`Member`
+            is in a guild message context. Sometimes it can be a :class:`User`
+            if the member has left the guild.
+        """
+
+        if not self._message or not self._state:  # Make type checker happy
+            raise RuntimeError('You cannot fetch users in a non-message-attached poll')
```

Comment:
```suggestion
            raise ClientException('You cannot fetch users to a poll not sent with a message')
```

Import `ClientException` and document this in `Raises`.